File: src/main/java/me/creepinson/creepinoutils/util/network/INetwork.java
package me.creepinson.creepinoutils.util.network;

import dev.throwouterror.util.math.Tensor;
import net.minecraft.nbt.CompoundNBT;
import net.minecraft.world.World;
import net.minecraftforge.common.util.INBTSerializable;

import java.io.Serializable;
import java.util.Set;

public interface INetwork<T> extends INBTSerializable<CompoundNBT>, Serializable {
    World getWorld();

    void refresh();

    Set<Tensor> getConnections();

    void merge(INetwork<T> net);

    void split(IBaseTile splitPoint);
}

File: src/main/kotlin/com/theoparis/creepinoutils/util/api/INetwork.kt
package com.theoparis.creepinoutils.util.api

import net.minecraft.nbt.CompoundNBT
import net.minecraft.util.math.vector.Vector3d
import net.minecraft.world.World
import net.minecraftforge.common.util.INBTSerializable
import java.io.Serializable

interface INetwork<T> : INBTSerializable<CompoundNBT>, Serializable {
    val world: World?
    fun refresh()
    val connections: Set<Vector3d>
    fun merge(net: INetwork<T>)
    fun split(splitPoint: IBaseTile)
}